webdictionary/src/main/java/com/eis0/kademlia/KademliaId.java

```diff
@@ -146,7 +172,48 @@ public KademliaId xor(KademliaId nid) {
         return new KademliaId(result);
     }
 
+    /**
+     * Generates a NodeId that is some distance away from this NodeId
+     *
+     * @param distance The index of the first bit that should be different,
+     *                 From 0 to 159, where 0 changes the most significant bit
+     * @return The newly generated NodeId with a given distance from this.
+     * @throws IllegalArgumentException if the distance is < 0 or >= 160
+     * @author Edordo Raimondi, improved by Marco Cognolato
+     */
+    public KademliaId generateNodeIdByDistance(int distance) {
+        if(distance < 0 || distance >= ID_LENGTH) throw new IllegalArgumentException();
+
+        byte[] result = keyBytes.clone();
+
+        // calculate the index of the byte and bit to update
+        int byteToUpdateIndex = (distance) / 8;
+        int bitToUpdateIndex = 7 - (distance % 8);
+
+        //change only the bit at the distance requested
+        result[byteToUpdateIndex] = (byte)(result[byteToUpdateIndex] ^ 0x01<<bitToUpdateIndex);
+        return new KademliaId(result);
+    }
 
+    /**
+     * Counts the number of leading 0's in this NodeId
+     *
+     * @return Integer representing the number of leading 0's,
+     * also returns -1 if there's only leading 0's
+     * @author EdoardoRaimondi, Marco Cognolato
+     */
+    public int getFirstSetBitIndex() {
```

Comment:
> As it is called now, this method name suggest returns the index of the first set bit index.

That's exactly what it does, it returns the index of the first bit = 1. I guess I'll update the specifications to avoid confusion 👍